core/src/main/scala/org/genivi/sota/core/db/Campaigns.scala

```diff
@@ -184,4 +220,46 @@ object Campaigns {
 
     dbIO.transactionally
   }
+
+  def setSize(id: Campaign.Id, size: Long)(implicit ec: ExecutionContext): DBIO[Unit] = {
+    campaignsMeta
+      .filter(_.id === id)
+      .map(_.size)
+      .update(Some(size))
+      .handleSingleUpdateError(MissingCampaign)
+  }
+
+  def setDeltaFrom(id: Campaign.Id, deltaFrom: Option[PackageId])(implicit ec: ExecutionContext): DBIO[Unit] = {
+    val dbIO = for {
+      _    <- canEdit(id)
+      cam  <- fetch(id)
+      _    <- if(deltaFrom.isDefined && cam.packageId.isDefined && deltaFrom.get.version == cam.packageId.get.version) {
+                DBIO.failed(InvalidDeltaFrom)
+              } else {
+                DBIO.successful(())
+              }
```

Comment:
Checking in both the UI and the backend is annoying, but I think belts and braces is appropriate here since it's a public API.